src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -109,39 +106,18 @@ public unsafe static SocketError Accept(SafeCloseSocket socketHandle, byte[] soc
 
         private void InnerReleaseHandle()
         {
-            // No-op for Unix.
+            if (_asyncContext != null)
+            {
+                _asyncContext.Close();
```

Comment:
Any value in setting it back to null after closing it?  Might it prevent double-closes, or is that handled internally by Close?  Or maybe there's an invariant that once non-null always non-null?